pkg/storage/redis/redis.go

```diff
@@ -257,7 +273,10 @@ func (db *DB) doNotifyLoop(ctx context.Context, ch chan struct{}, psc *redis.Pub
 			case <-ctx.Done():
 				log.Warn().Err(ctx.Err()).Msg("context done, stop receive from redis channel")
 				return
-			case ch <- struct{}{}:
+			default:
+				db.notifyChMu.Lock()
+				ch <- struct{}{}
```

Comment:
this can block and leave the notify mutex permanently locked

Reply:
In theory, yes. In practice, `doNotifyLoop` is the only one that send data to the channel, so it can't block.

Reply:
If you worried about that, then I think we can add another select case to ctx.Done(), PR is coming.

Reply:
@calebdoxsey fixed.